### DISCUSSION TITLE
Valgrind build for Axis ACAP SDK v2

### DISCUSSION THREAD

**Discussion by dtsmith2001** [AxisCommunications/discussions#151] [ACAP] [answered]
There are supported cameras running the MIPS CPU.

- I'd like to know if valgrind can be built by the MIPS toolchain for SDK v2.
- If not, what about the Address Sanitizer for SDK v2.

I can attach `gdb-mips` to debug remotely. However, I don't get any stack trace information at all, even though my ACAP application is built with `-g3`.

I have also added signal handlers for SEGV and ABRT to my ACAP application code. I'm not getting anything meaningful using Boost::Stacktrace.

I'm looking for any advice on how to troubleshoot on the MIPS cameras.

**Comment by alphaon** [comment, marked as answer]
Try to recode your app with Rust, there is a crate with valgrind.
